documentation/src/docs/basic-reader-and-writer.md

```diff
@@ -9,243 +9,204 @@ You may obtain a copy of the License at
-that sets the minimum degree of read parallelism to be maintained; if this value
-is set at 3, for example, there will always be 3 Stream Segments available on
-the Stream.  Currently, this property is effective only when the stream is
+objects in Pravega use a builder pattern for construction. [**Retention Policy**](pravega-concepts.md#retention-policy) and [**Scaling
+Policy**](pravega-concepts.md#elastic-streams-auto-scaling) are the two important configuration items related to streams.   
+
+The minimum number of Segments is a factor that sets the minimum degree of read parallelism to be maintained; for example if this value is set as three, there will always be three Stream Segments available on the Stream.  Currently, this property is effective only when the stream is
 created; at some point in the future, update stream will allow this factor to be
 used to change the minimum degree of read parallelism on an existing Stream.
 
-Once the `StreamConfiguration` object is created, creating the Stream is straight
-forward (line 8).  After the Stream is created, we are all set to start writing
+Once the Stream Configuration (`streamConfig`) object is built, creating the Stream is straight
+forward using `createStream()`. After the Stream is created, we are all set to start writing
 Event(s) to the Stream.
 
+
 ## Writing an Event using EventWriter
 
 Applications use an `EventStreamWriter` object to write Events to a Stream.  The
-key object to creating the `EventStreamWriter` is the ClientFactory.  The
-ClientFactory is used to create Readers, Writers and other types of Pravega
+key object to creating the `EventStreamWriter` is the `ClientFactory`.  The
+`ClientFactory` is used to create Readers, Writers and other types of Pravega
 Client objects such as the State Synchronizer (see [Working with Pravega: State
 Synchronizer](state-synchronizer.md)).
 
-Line 10 shows the creation of a ClientFactory.  A ClientFactory is created in
-the context of a Scope, since all Readers, Writers and other Clients created by
-the ClientFactory are created in the context of that Scope.  The ClientFactory
-also needs a URI to one of the Pravega Controllers, just like StreamManager.
+A `ClientFactory` is created in the context of a Scope, since all Readers, Writers and other Clients created by the `ClientFactory` are created in the context of that Scope. The `ClientFactory`
+also needs a URI to one of the Pravega Controllers (`ClientFactory.withScope(scope, controllerURI)`) , just like `StreamManager`.
 
-Because ClientFactory and the objects it creates consumes resources from
-Pravega, it is a good practice to create these objects in a try-with-resources
-statement.  Because ClientFactory and the objects it creates all implement
-Autocloseable, the try-with-resources approach makes sure that regardless of how
-your application ends, the Pravega resources will be properly closed in the
+As the `ClientFactory` and the objects it creates consumes resources from
```

Comment:
Please, rephrase:
`As the ClientFactory and the objects it creates consume resources from Pravega and implement `AutoCloseable`, it is a good practice to create these objects using a try-with-resources. By doing this, we make sure that, regardless of how the application ends, the Pravega resources will be properly closed in the right order.`

Reply:
Accepted suggestion